VMBackup/main/fsfreezer.py

```diff
@@ -172,7 +176,49 @@ def freeze_safe(self,timeout):
                 self.logger.enforce_local_flag(True)
             else:
                 self.logger.enforce_local_flag(False) 
-            sig_handle=self.freeze_handler.startproc(args)
+
+            start_time = datetime.datetime.utcnow()
+
+            while self.getLockRetry < self.maxGetLockRetry:
+                try:
+                    os.mkdir('/etc/azure/MicrosoftRecoverySvcsSafeFreezeLock')
+                    file = open("/etc/azure/MicrosoftRecoverySvcsSafeFreezeLock/SafeFreezeLockFile","w")
+                    self.logger.log("/etc/azure/MicrosoftRecoverySvcsSafeFreezeLock/SafeFreezeLockFile file opened Sucessfully",True)
+                    try:
+                        fcntl.lockf(file, fcntl.LOCK_EX | fcntl.LOCK_NB)
+                        self.logger.log("Aquiring lock succeeded",True)
+                        self.isAquireLockSucceeded = True
+                        break
+                    except Exception as ex:
+                        file.close()
+                        raise ex
+                except Exception as e:
```

Comment:
Also delete the file on exception

Reply:
moved final delete outside if so delete will always happen